### DISCUSSION TITLE
Error: P1001: Can't reach database server

### DISCUSSION THREAD

**Discussion by MrDiamondDog** [supabase/supabase#21576] [Questions] [answered]
No matter what I try, I can not get my Prisma app to connect to my supabase database.

I get the error in the title whenever I run `npx prisma migrate dev --name init`

I have tried many solutions and none of them work, including remaking the entire database.

I have also tried connecting using pgAdmin, and it didn't work.

My URLs are as follows:
- `postgres://[db-user].[project-ref]:[db-password]@aws-0-[aws-region].pooler.supabase.com:6543/[db-name]?pgbouncer=true&connection_limit=1`
- `postgres://[db-user].[project-ref]:[db-password]@aws-0-[aws-region].pooler.supabase.com:5432/[db-name]`

My schema.prisma is
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}

[tables]
```

**Comment by GaryAustin1** [comment]
Do your final URL's look somthing like this: `postgres://postgres.udkfjdskcccssdeeee:MYPassword@aws-0-us-east-1.pooler.supabase.com:5432/postgres`

**Reply by MrDiamondDog** [reply]
They look just like that, yes

**Reply by GaryAustin1** [reply]
Does your password have any special characters in it that need to be escaped?
https://github.com/orgs/supabase/discussions/17817 has most of the latest issues and solutions.

**Reply by MrDiamondDog** [reply]
Just letters.

**Comment by MrDiamondDog** [comment, marked as answer]
Wait at least an hour after you make the database. I also deleted the migrations folder (not sure if that was the actual solution). Also make sure you are on an IPv6 network.
